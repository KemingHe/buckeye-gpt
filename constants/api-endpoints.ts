// Langchain OpenAI endpoints.
export const LANGCHAIN_OPENAI_LITE_API_ENDPOINT: string =
  '/api/langchain/openai/lite';
export const LANGCHAIN_OPENAI_REGULAR_API_ENDPOINT: string =
  '/api/langchain/openai/regular';
export const LANGCHAIN_OPENAI_PREMIUM_API_ENDPOINT: string =
  '/api/langchain/openai/premium';

// Langchain Google Gemini endpoints.
export const LANGCHAIN_GOOGLE_GEMINI_LITE_API_ENDPOINT: string =
  '/api/langchain/gemini/lite';
export const LANGCHAIN_GOOGLE_GEMINI_REGULAR_API_ENDPOINT: string =
  '/api/langchain/gemini/regular';
export const LANGCHAIN_GOOGLE_GEMINI_PREMIUM_API_ENDPOINT: string =
  '/api/langchain/gemini/premium';

// Langchain Anthropic Claude endpoints.
export const LANGCHAIN_ANTHROPIC_CLAUDE_LITE_API_ENDPOINT: string =
  '/api/langchain/claude/lite';
export const LANGCHAIN_ANTHROPIC_CLAUDE_REGULAR_API_ENDPOINT: string =
  '/api/langchain/claude/regular';
export const LANGCHAIN_ANTHROPIC_CLAUDE_PREMIUM_API_ENDPOINT: string =
  '/api/langchain/claude/premium';
