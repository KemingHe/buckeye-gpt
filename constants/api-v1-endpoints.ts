const prefix: string = '/api/v1';

// Auth endpoints.
export const AUTH_MAGIC_LINK_SEND_API_ENDPOINT: string = `${prefix}/auth/magic-link/send`;

// Chats endpoints.
// OpenAI GPT endpoints.
export const LANGCHAIN_OPENAI_GPT_LITE_API_ENDPOINT: string = `${prefix}/chats/openai/lite`;
export const LANGCHAIN_OPENAI_GPT_REGULAR_API_ENDPOINT: string = `${prefix}/chats/openai/regular`;
export const LANGCHAIN_OPENAI_GPT_PREMIUM_API_ENDPOINT: string = `${prefix}/chats/openai/premium`;

// Google Gemini endpoints.
export const LANGCHAIN_GOOGLE_GEMINI_LITE_API_ENDPOINT: string = `${prefix}/chats/gemini/lite`;
export const LANGCHAIN_GOOGLE_GEMINI_REGULAR_API_ENDPOINT: string = `${prefix}/chats/gemini/regular`;
export const LANGCHAIN_GOOGLE_GEMINI_PREMIUM_API_ENDPOINT: string = `${prefix}/chats/gemini/premium`;

// Anthropic Claude endpoints.
export const LANGCHAIN_ANTHROPIC_CLAUDE_LITE_API_ENDPOINT: string = `${prefix}/chats/claude/lite`;
export const LANGCHAIN_ANTHROPIC_CLAUDE_REGULAR_API_ENDPOINT: string = `${prefix}/chats/claude/regular`;
export const LANGCHAIN_ANTHROPIC_CLAUDE_PREMIUM_API_ENDPOINT: string = `${prefix}/chats/claude/premium`;
