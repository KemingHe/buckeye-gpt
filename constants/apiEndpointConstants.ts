// Langchain endpoints.
export const LANGCHAIN_OPENAI_API_ENDPOINT: string = '/api/langchain/openai';

// Stack-Auth endpoints.
export const STACH_AUTH_SIGN_IN_API_ENDPOINT: string =
  '/api/stack/auth/sign-in';
