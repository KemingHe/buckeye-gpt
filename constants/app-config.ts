// Color constants.
export const COLOR_PRIMARY: string = '#1d242a';

// Preview image constants.
export const SOCIALIFY_PREVIEW_IMAGE_URL: string =
  'https://socialify.git.ci/KemingHe/buckeye-gpt/png?description=1&forks=1&issues=1&language=1&logo=https%3A%2F%2Fgithub.com%2FKemingHe%2Fbuckeye-gpt%2Fblob%2Fmain%2Fpublic%2Fimages%2Ftransparent-buckeye-gpt-icon-512x512px.png%3Fraw%3Dtrue&name=1&owner=1&pattern=Plus&pulls=1&stargazers=1&theme=Light';
export const SOCIALIFY_PREVIEW_IMAGE_WIDTH: number = 1280;
export const SOCIALIFY_PREVIEW_IMAGE_HEIGHT: number = 640;
export const SOCIALIFY_PREVIEW_IMAGE_ALT: string =
  'Buckeye GPT is a open-source multi-modal ChatGPT clone with RAG built free for students at the Ohio State University.';
