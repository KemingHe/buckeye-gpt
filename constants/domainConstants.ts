// Development domains.
export const LOCAL_NETWORK_DOMAIN: string = 'http://192.168.1.67:3000';
export const LOCALHOST_DOMAIN: string = 'http://localhost:3000';

// Production domain.
export const PRODUCTION_DOMAIN: string = 'https://buckeye.chat';
