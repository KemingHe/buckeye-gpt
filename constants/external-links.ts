// Developer links.
export const GITHUB_REPO_LINK: string =
  'https://github.com/KemingHe/buckeye-gpt';
export const DEVELOPER_GITHUB_LINK: string = 'https://github.com/KemingHe';
export const DEVELOPER_LINKEDIN_LINK: string =
  'https://www.linkedin.com/in/keminghe';

// Legal document links.
export const LEGAL_DISCLAIMER_LINK: string =
  'https://github.com/KemingHe/buckeye-gpt/blob/main/docs/legal/legal-disclaimer.md';
export const PRIVACY_POLICY_LINK: string =
  'https://github.com/KemingHe/buckeye-gpt/blob/main/docs/legal/privacy-policy.md';
export const TERMS_AND_CONDITIONS_LINK: string =
  'https://github.com/KemingHe/buckeye-gpt/blob/main/docs/legal/terms-and-conditions.md';
