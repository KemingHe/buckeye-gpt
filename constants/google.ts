export const MEASUREMENT_ID: string = process.env
  .NEXT_PUBLIC_GOOGLE_ANALYTICS_MEASUREMENT_ID as string;
