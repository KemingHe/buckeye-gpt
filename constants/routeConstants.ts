export const HOME_ROUTE = '/';

// Stack Auth routes.
export const AUTH_SIGN_IN_ROUTE = '/auth/sign-in';
export const AUTH_VERIFY_EMAIL_ROUTE = '/auth/verify-email';
