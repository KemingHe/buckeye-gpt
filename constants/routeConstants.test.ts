import * as routeConstants from '@/constants/routeConstants';

describe('Route constants', (): void => {
  it('matches the snapshot', (): void => {
    expect(routeConstants).toMatchSnapshot();
  });
});
