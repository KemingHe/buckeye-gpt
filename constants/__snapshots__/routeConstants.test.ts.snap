// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Route constants > matches the snapshot 1`] = `
{
  "AUTH_SIGN_IN_ROUTE": "/auth/sign-in",
  "AUTH_VERIFY_EMAIL_ROUTE": "/auth/verify-email",
  "HOME_ROUTE": "/",
}
`;
